src/View/Helper/PaginatorHelper.php

```diff
@@ -1169,4 +1170,43 @@ public function implementedEvents()
     {
         return [];
     }
+
+    /**
+     * Dropdown array for Select limit
+     *
+     * @param array $limits This is array of options.
+     * @param int $default Default limit for option selecting. Default is Paginator limit.
+     * @param array $options Options for Select tag attributes like class, id or event
+     *
+     * @return string html output.
+     */
+    public function limitSelect($limits = [], $default = null, $options = [])
+    {
+        $out = $this->Form->create(null, ['type' => 'get']);
+
+        if (empty($limits) || !is_array($limits)) {
+            $limits = [
+                20 => '20',
+                25 => '25',
+                50 => '50',
+                100 => '100'
+            ];
+        }
+
+        if (empty($default) || !is_integer($default)) {
+            $default = $this->param('perPage');
+        }
+
+        if (!in_array($default, $limits)) {
+            $limits += [$default => $default];
+        }
+
+        ksort($limits);
+
+        $out .= $this->Form->input('limit', ($options + ['type' => 'select', 'label' => __("View"), 'value' => $default, 'default' => $default, 'options' => $limits, 'onChange' => 'this.form.submit()']));
```

Comment:
You should not set value IMO, but only default. Otherwise you hardcode it and one cannot change it anymore at runtime.